### PR TITLE
Frontend dashboard page

### PR DESCRIPTION
**Pull request by cjdenio**

### CODE REVIEW

**Comment by cjdenio** [comment]
Relevant screenshot:

Screen Shot 2021-04-28 at 5 20 18 PM

File: frontend/layouts/dashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+}: {
+  title?: string;
+  items: { image?: string; icon?: string; text: string }[];
+}) {
+  return (
+    <Box mt={4}>
+      {title && <Heading mb={3}>{title}</Heading>}
+      {items.map((v, i) => {
+        return (
+          <SidebarItem key={i} icon={v.icon} image={v.image}>
```
Comment (anirudhb) [inline frontend/layouts/dashboard.tsx]: Key by text since index is not unique

File: frontend/layouts/dashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+export default function DashboardLayout({
+  title,
+  sidebarSections,
+  children,
+}: PropsWithChildren<{ title: string; sidebarSections: SidebarSection[] }>) {
+  const { data: user } = useSWR("/users/me", fetchApi);
+
+  return (
+    <>
```
Comment (anirudhb) [inline frontend/layouts/dashboard.tsx]: Don't see the need for this `<>` since there's only one inner element

File: frontend/pages/dashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  const { data: user } = useSWR("/users/me", fetchApi);
+  const { data: teams } = useSWR("/users/me/teams", fetchApi);
+  const { data: apps } = useSWR("/users/me/apps", fetchApi);
+
+  // Get personal apps
+  const personalApps =
+    teams && apps
+      ? apps.apps.filter(
+          (app: any) =>
+            app.TeamID == teams.teams.find((team) => team.Personal).ID
```
Comment (anirudhb) [inline frontend/pages/dashboard.tsx]: This is O(n^2), store the personal team ID in a variable

File: frontend/pages/dashboard.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        )
+      : null;
+
+  const sidebarSections: SidebarSection[] = [
+    {
+      items: [
+        {
+          image: user?.user.Avatar,
```
Comment (anirudhb) [inline frontend/pages/dashboard.tsx:42]: Handle the logged out case more properly

File: frontend/pages/dashboard.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            }))
+        : [],
+    },
+  ];
+
+  return (
+    <DashboardLayout title="Personal Apps" sidebarSections={sidebarSections}>
+      {personalApps &&
+        personalApps.map((v, i) => {
+          return <App name={v.Name} shortName={v.ShortName} key={i} />;
```
Comment (anirudhb) [inline frontend/pages/dashboard.tsx]: Prefer to use app id as key since index is not unique

**Review by anirudhb — CHANGES_REQUESTED** [review]
lgtm, nits

**Review by anirudhb — APPROVED** [review]
lgtm

**Pull request merged**